src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs

```diff
@@ -442,6 +479,10 @@ public int MaxModelValidationErrors
             }
         }
 
+        internal event Action<MvcOptions> OnAfterPostConfigure;
+
+        internal void InvokeOnAfterPostConfigure() => OnAfterPostConfigure?.Invoke(this);
```

Comment:
:(

Reply:
Based on our offline conversation, I tried making an `MvcXmlOptions` type and putting the compat flag on it, setting up the xml formatters after the compat option was initialized. However, since nothing uses the options type, the configure never executes. Leaving the flag on MvcOptions seems like the least problematic way forward. However it also keeps this `event` around and makes you 😭.